tests/test_permissions_synchronizer.py

```diff
@@ -20,82 +20,81 @@ class TestSyncPermissions(unittest.TestCase):
+        mapping_point_1 = "mapping_point_1"
+        cls.mapped_service = {service1: service2,
                               service2: service1}
-
-            data = {
-                "services": {
-                    "Magpie": {"active": True},
-                    service1: {"active": True},
-                    service2: {"active": True}
-                },
-                "sync_permissions": {
-                    mapping_point_1: {
-                        "services": {
-                            service1: res_root[service1],
-                            service2: res_root[service2]
-                        },
-                        "permissions_mapping": [
-                            {
-                                service1: sync_perm_name[service1],
-                                service2: sync_perm_name[service2]
-                            }
-                        ]
-                    }
+        data = {
+            "services": {
+                "Magpie": {"active": True},
+                service1: {"active": True},
+                service2: {"active": True}
+            },
+            "sync_permissions": {
+                mapping_point_1: {
+                    "services": {
+                        service1: cls.res_root[service1],
+                        service2: cls.res_root[service2]
+                    },
+                    "permissions_mapping": [
+                        {
+                            service1: cls.sync_perm_name[service1],
+                            service2: cls.sync_perm_name[service2]
+                        }
+                    ]
                 }
             }
-            tmp.write(yaml.safe_dump(data))
-            tmp.seek(0)  # back to start since file still open (auto-delete if closed)
-            with contextlib.ExitStack() as stack:
-                stack.enter_context(mock.patch("cowbird.services.impl.magpie.Magpie",
-                                               side_effect=utils.MockMagpieService))
-                stack.enter_context(mock.patch("cowbird.services.impl.geoserver.Geoserver",
-                                               side_effect=utils.MockAnyService))
-                stack.enter_context(mock.patch("cowbird.services.impl.thredds.Thredds",
-                                               side_effect=utils.MockAnyService))
-                utils.get_test_app(settings={"cowbird.config_path": tmp.name})
 
-                resource_name = "resource1"
-                magpie = ServiceFactory().get_service("Magpie")
+        }
+        cls.cfg_file = tempfile.NamedTemporaryFile(mode="w", suffix=".cfg", delete=False)
+        with cls.cfg_file as f:
+            f.write(yaml.safe_dump(data))
+        cls.app = utils.get_test_app(settings={"cowbird.config_path": cls.cfg_file.name})
+
+    @classmethod
+    def tearDownClass(cls):
+        utils.clear_services_instances()
+        os.unlink(cls.cfg_file.name)
+
+    def test_sync(self):
```

Comment:
A small docstring that indicates the purpose & expected outcome of this test will help understand what is being validated here.